lib/models/WebsiteConfiguration.ts

```diff
@@ -0,0 +1,215 @@
+export type Redirect = {
+    protocol?: string;
+    hostName?: string;
+    replaceKeyPrefixWith?: string;
+    replaceKeyWith?: string;
+    httpRedirectCode: string;
+};
+export type Condition = {
+    keyPrefixEquals?: string;
+    httpErrorCodeReturnedEquals?: string;
+};
+export type RoutingRuleParams = { redirect: Redirect; condition?: Condition };
+
+export class RoutingRule {
+    _redirect?: Redirect;
+    _condition?: Condition;
+
+    /**
+    * Represents a routing rule in a website configuration.
+    * @constructor
+    * @param params - object containing redirect and condition objects
+    * @param params.redirect - specifies how to redirect requests
+    * @param [params.redirect.protocol] - protocol to use for redirect
+    * @param [params.redirect.hostName] - hostname to use for redirect
+    * @param [params.redirect.replaceKeyPrefixWith] - string to replace
+    *   keyPrefixEquals specified in condition
+    * @param [params.redirect.replaceKeyWith] - string to replace key
+    * @param [params.redirect.httpRedirectCode] - http redirect code
+    * @param [params.condition] - specifies conditions for a redirect
+    * @param [params.condition.keyPrefixEquals] - key prefix that
+    *   triggers a redirect
+    * @param [params.condition.httpErrorCodeReturnedEquals] - http code
+    *   that triggers a redirect
+    */
+    constructor(params?: RoutingRuleParams) {
+        if (params) {
```

Comment:
Is params optional? The JSDoc has it as required but the JS code has this check to

Reply:
I prefered to stick with the implementation because the JSDoc was sometimes outdated. The type is mainly to reflect the construct.